src/node_file.cc

```diff
@@ -138,9 +124,27 @@ void FSReqWrap::Resolve(Local<Value> value) {
   MakeCallback(env()->oncomplete_string(), arraysize(argv), argv);
 }
 
+void FSReqWrap::Init(const char* syscall,
+                     const char* data,
+                     enum encoding encoding,
+                     Ownership ownership) {
+  CHECK_EQ(data_, nullptr);  // Initialize only once
+  syscall_ = syscall;
+  encoding_ = encoding;
+
+  const bool copy = (data != nullptr && ownership == Ownership::COPY);
+  if (copy) {
+    size_t size = 1 + strlen(data);
+    data_ = static_cast<char*>(memcpy(Calloc<char>(size), data, size));
```

Comment:
`Malloc(size)` should work just the same and doesn’t require zero-ing out memory that is being overwritten anyway

Also, these two lines are almost exactly reimplementing `strdup()`.

Reply:
good point about strdup :-)